app/src/main/java/org/openobservatory/ooniprobe/test/suite/AbstractSuite.java

```diff
@@ -14,154 +14,171 @@
+        this.title = title;
+        this.cardDesc = cardDesc;
+        this.icon = icon;
+        this.icon_24 = icon_24;
+        this.color = color;
+        this.themeLight = themeLight;
+        this.themeDark = themeDark;
+        this.desc1 = desc1;
+        this.anim = anim;
+        this.name = name;
+        this.dataUsage = dataUsage;
+    }
+
+    @CallSuper
+    public AbstractTest[] getTestList(@Nullable PreferenceManager pm) {
+        return testList;
+    }
+
+    public void setTestList(AbstractTest... tests) {
+        this.testList = tests;
+    }
+
+    public Result getResult() {
+        if (result == null)
+            result = new Result(name);
+        return result;
+    }
+
+    public void setResult(Result result) {
+        this.result = result;
+    }
+
+    public int getTitle() {
+        return title;
+    }
+
+    public int getCardDesc() {
+        return cardDesc;
+    }
+
+    public int getIcon() {
+        return icon;
+    }
+
+    public int getIconGradient() {
+        if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.M) {
+            return icon;
+        } else {
+            return icon_24;
+        }
+    }
+
+    public String getAnim() {
+        return anim;
+    }
+
+    public int getColor() {
+        return color;
+    }
+
+    public int getThemeLight() {
+        return themeLight;
+    }
+
+    public int getThemeDark() {
+        return themeDark;
+    }
+
+    public int getDesc1() {
+        return desc1;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public int getDataUsage() {
+        return dataUsage;
+    }
+
+    public Integer getRuntime(PreferenceManager pm) {
+        Integer runtime = 0;
+        for (AbstractTest test : getTestList(pm))
+            runtime += test.getRuntime(pm);
+        //TODO convert seconds to minutes and hours when needed
+        //if getRuntime <= MAX_RUNTIME_DISABLED show one hour
+        if (runtime <= PreferenceManager.MAX_RUNTIME_DISABLED)
+            runtime = 3600;
+        return runtime;
+    }
+
+    public ArrayList<AbstractSuite> asArray() {
+        ArrayList<AbstractSuite> list = new ArrayList<>();
+        list.add(this);
+        return list;
+    }
+
+    public boolean isTestEmpty(PreferenceManager preferenceManager) {
+        getTestList(preferenceManager);
+        return testList == null || testList.length == 0;
+    }
+
+    public boolean getAutoRun() {
+        return autoRun;
+    }
+
+    public void setAutoRun(boolean autoRun) {
+        this.autoRun = autoRun;
+    }
+
+    public static AbstractSuite getSuite(Application app,
+                                         String tn,
+                                         @Nullable List<String> urls,
+                                         String origin) {
+        for (AbstractSuite suite : TestAsyncTask.getSuites())
```

Comment:
Please, always use `{` and `}` even if there is just a single statement afterwards.